src/bin/ion/commands/stats.rs

```diff
@@ -34,81 +34,112 @@ impl IonCliCommand for StatsCommand {
-    writeln!(writer, "Symbols: {} ", out.symbols_count)
-        .expect("There is an error occurred while writing the symbols_count.");
-    writeln!(writer, "Local symbol tables: {} ", out.symtab_count)
-        .expect("There is an error occurred while writing the symtab_count.");
-    writeln!(writer, "Maximum container depth: {}", out.max_depth)
-        .expect("There is an error occurred while writing the max_depth.");
+    let histogram = plot::Histogram::new(&stats.size_vec, options);
+
+    if args.get_flag("count") {
+        writeln!(writer, "{}", stats.size_vec.len())?;
+        return Ok(());
+    } else {
+        writeln!(
+            writer,
+            "'samples' is the number of top-level values for the input stream."
+        )?;
+        writeln!(writer, "The unit of min, max, and avg size is bytes.")?;
+        writeln!(writer, "{}", histogram)?;
+        writeln!(writer, "Symbols: {} ", stats.symbols_count)?;
+        writeln!(writer, "Local symbol tables: {} ", stats.symtab_count)?;
+        writeln!(writer, "Maximum container depth: {}", stats.max_depth)?;
+        if stats.unparseable_count > 0 {
+            writeln!(writer, "Unparseable values: {}", stats.unparseable_count)?;
+        }
+    }
+
     Ok(())
 }
 
-fn analyze_data_stream<Input: IonInput>(reader: &mut SystemReader<AnyEncoding, Input>) -> Output {
+fn analyze_data_stream<Input: IonInput>(
+    reader: &mut SystemReader<AnyEncoding, Input>,
+) -> StreamStats {
     let mut size_vec: Vec<f64> = Vec::new();
     let mut symtab_count = 0;
     let mut max_depth = 0;
+    let mut unparseable_count = 0;
+
     loop {
-        let system_value = reader.next_item().unwrap();
-        match system_value {
-            SystemStreamItem::SymbolTable(_) => {
-                symtab_count += 1;
-            }
-            SystemStreamItem::Value(value) => {
-                let size = system_value.raw_stream_item().unwrap().span().bytes().len();
-                size_vec.push(size as f64);
-                let current_depth = top_level_max_depth(value);
-                max_depth = max(max_depth, current_depth);
-            }
-            SystemStreamItem::EndOfStream(_) => {
-                break;
+        let system_result = reader.next_item();
+        use SystemStreamItem::*;
+
+        match system_result {
+            Err(e) => {
+                unparseable_count += 1;
+                if matches!(e, IonError::Incomplete(..)) {
+                    break;
+                }
             }
-            _ => {}
+
+            Ok(item) => match item {
+                EndOfStream(_) => break,
+                VersionMarker(_) | EncodingDirective(_) => continue,
+                SymbolTable(_) => symtab_count += 1,
+                system_value @ Value(raw_value) => {
+                    let size = system_value.raw_stream_item().unwrap().span().bytes().len();
```

Comment:
In Ion 1.1, you can have system values that are not backed by a raw stream item.

Reply:
If it's an ephemeral/computed value, then it doesn't have a size. You can choose whether that means `size=0` bytes or if you don't even count it. The `unwrap()` will cause a panic, though.